cedar-policy-core/src/validator/coreschema.rs

```diff
@@ -260,6 +246,41 @@ impl ast::RequestSchema for ValidatorSchema {
         }
         Ok(())
     }
+
+    /// Validate a context against a schema for a specific action
+    fn validate_context<'a>(
+        &self,
+        context: &ast::Context,
+        action: &ast::EntityUID,
+        extensions: &Extensions<'a>,
+    ) -> std::result::Result<(), RequestValidationError> {
+        // Following the same logic in validate_request
+        // Get the action ID
+        let action_arc = Arc::new(action.clone());
```

Comment:
can we avoid cloning the action and constructing an `Arc` on the happy path?  I think the clone and Arc are only needed on error paths